code/datums/Emote_system/emote.dm

```diff
@@ -0,0 +1,573 @@
+		user.set_typing_indicator(1)
+		user.hud_typing = 1
+		params["num"] = getNumber(user)
+		user.hud_typing = 0
+		user.set_typing_indicator(0)
+
+	if(canTarget)
+		user.set_typing_indicator(1)
+		user.hud_typing = 1
+		params["target"] = getTarget(user)
+		user.hud_typing = 0
+		user.set_typing_indicator(0)
+	return params
+
+// return INVALID from either of these getters if you've tested the input and it's failed
+/datum/emote/proc/getNumber(var/mob/user)
+	var/number = input("How many?", "Enter number") as null|num
+	return number
+
+/datum/emote/proc/getTarget(var/mob/user)
+	if(user.sdisabilities & BLIND || user.blinded || user.paralysis)
+		return
+	var/list/targets = list()
+	if(!mustTarget)
+		targets += "None"
+	var/target
+	if(targetMob)
+		target = getMobTarget(user, targets)
+	else
+		target = getAtomTarget(user, targets)
+	if(!target)
+		return INVALID
+	if(target == "None")
+		return
+	return target
+
+/datum/emote/proc/getMobTarget(var/mob/user, var/list/targets)
+	for(var/mob/M in view(getLoc(user)))
+		targets += M
+	var/mob/target = input("Select target", "Target Mob") as null|anything in targets
+	return target
+
+
+/datum/emote/proc/getAtomTarget(var/mob/user, var/list/targets)
+	for(var/A in oview(getLoc(user)))
+		if(ismob(A) || isobj(A) || isturf(A))
+			targets += A
+	var/atom/target = input("Select target", "Target") as null|anything in targets
+	return target
+
+// returns the reason the user can't currently do the emote
+/datum/emote/proc/prevented(var/mob/user)
+	if(user.stat == DEAD)
+		return "you are dead"
+	if(user.stat == UNCONSCIOUS)
+		return "you are unconscious"
+	if(restrained && user.restrained())
+		return "you are restrained"
+	if(isbrain(user))
+		var/mob/living/carbon/brain/brain = user
+		if(!(brain.container && istype(brain.container, /obj/item/device/mmi)))
+			return "you need to be in an mmi to do this"
+
+// return 1 if this emote can be used by this type of user
+/datum/emote/proc/available(var/mob/user)
+	return
+
+/datum/emote/proc/createMessage(var/mob/user, var/list/params)
+	if(!text)
+		return
+	var/message = ""
+
+	if(doMime(user))
+		message = mimeMessage(user, params)
+		if(message)
+			return message
+
+	if(muzzleAffected && isMuzzled(user))
+		message = muzzleMessage(user, params)
+		return message
+
+	if(checkForParams(params))
+		message = paramMessage(user, params)
+		return message
+
+	message = standardMessage(user, params)
+
+	return message
+
+/datum/emote/proc/isMuzzled(var/mob/user)
+	if(user.sdisabilities & MUTE || user.is_muzzled())
+		return 1
+	if(!isliving(user))
+		return
+	var/mob/living/L = user
+	if (L.silent)
+		return 1
+
+/datum/emote/proc/checkForParams(var/list/params)
+	for(var/p in params)
+		if(p == "target")
+			continue
+		return 1
+
+/datum/emote/proc/addExtras(var/mob/user, var/list/params, var/message = "")
+	if(!message)
+		return
+	if(startText)
+		message = "[startText] [message]"
+	message = "<span class = '[emoteSpanClass]'>[message]</span>"
```

Comment:
Current standard is no space between `span class`, `=` and the class. There are a lot of these.